### PR TITLE
Adding documentation on Adapter.

### PR DESCRIPTION
**Pull request by maximelb**
## Description of the change

Adding/updating/refactoring doc for the lc-adapter.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: docs/lc_adapter.md [hunk 1]
```diff
@@ -10,7 +10,25 @@ The LimaCharlie Adapter can be used to ingest external data streams from many di
 ## Availability
 
 * Docker: https://hub.docker.com/r/refractionpoint/lc-adapter
-* Various Executables: TBD
+* Various Executables:
+  * [Windows x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_windows_64_1.2.2.exe)
```
Comment (amrikr) [inline docs/lc_adapter.md:14]:
```suggestion
  * [Windows x64](https://app.limacharlie.io/get/adapter/win64)
```

File: docs/lc_adapter.md [hunk 2]
```diff
@@ -10,7 +10,25 @@ The LimaCharlie Adapter can be used to ingest external data streams from many di
...
+  * [macOS x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_macos_64_1.2.2)
```
Comment (amrikr) [inline docs/lc_adapter.md:15]:
```suggestion
  * [macOS x64](https://app.limacharlie.io/get/adapter/mac64)
```

File: docs/lc_adapter.md [hunk 3]
```diff
@@ -10,7 +10,25 @@ The LimaCharlie Adapter can be used to ingest external data streams from many di
...
+  * [macOS arm64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_macos_arm64_1.2.2)
```
Comment (amrikr) [inline docs/lc_adapter.md:16]:
```suggestion
  * [macOS arm64](https://app.limacharlie.io/get/adapter/macarm64)
```

File: docs/lc_adapter.md [hunk 4]
```diff
@@ -10,7 +10,25 @@ The LimaCharlie Adapter can be used to ingest external data streams from many di
...
+  * [Linux x64](https://storage.googleapis.com/limacharlie-io/installers/lc_adapter_linux_64_1.2.2)
```
Comment (amrikr) [inline docs/lc_adapter.md:17]:
```suggestion
  * [Linux x64](https://app.limacharlie.io/get/adapter/linux64)
```

File: docs/lc_adapter.md [hunk 5]
```diff
@@ -10,7 +10,25 @@ The LimaCharlie Adapter can be used to ingest external data streams from many di
...
+
+## Concept
+The LimaCharlie Adapter is a piece of software that forwards any logs or telemetry of your choice to LimaCharlie in real-time.
+It can be run on any platform and in any location (on premises for example).
+
+The Adapter may itself get the logs/telemetry any number of locations and using many different methods like:
+
+* Syslog
+* AWS S3
+* Google Cloud Pubsub
```
Comment (amrikr) [inline docs/lc_adapter.md:28]:
```suggestion
* Google Cloud Pub/Sub
```

File: docs/lc_adapter.md [hunk 6]
```diff
@@ -47,42 +77,231 @@ Other common configurations that are not required include:
...
+
+To test it, assuming we're on the same Debian box as the container, pipe the syslog to the container:
 ```
 journalctl -f -q | netcat 127.0.0.1 4444
 ```
 
-### S3
+### CarbonBlack from S3
+
+This example shows connecting CarbonBlack sensors from data exported by the CarbonBlack API to an S3 bucket.
```
Comment (amrikr) [inline docs/lc_adapter.md:249]:
```suggestion
This example shows connecting Carbon Black sensors from data exported by the Carbon Black API to an S3 bucket.
```

File: docs/lc_adapter.md [hunk 7]
```diff
@@ -47,42 +77,231 @@ Other common configurations that are not required include:
...
+./lc_adapter s3 client_options.identity.installation_key=e9a3bcdf-efa2-47ae-b6df-579a02f3a54d client_options.identity.oid=8cbe27f4-bfa1-4afb-ba19-138cd51389cd client_options.platform=carbon_black client_options.sensor_seed_key=tests3 bucket_name=lc-cb-test access_key=YYYYYYYYYY secret_key=XXXXXXXX  "prefix=events/org_key=NKZAAAEM/"
 ```
 
+Here's a breakdwn of the above example:
+
+* `lc_adapter`: simply the CLI Adapter.
+* `s3`: the data will be collected from an AWS S3 bucket.
+* `client_options.identity.installation_key=....`: the installation key value from LimaCharlie.
+* `client_options.identity.oid=....`: the Organization ID from LimaCharlie the installation key above belongs to.
+* `client_options.platform=carbon_black`: this indicates the data received will be CarbonBlack events from their API.
```
Comment (amrikr) [inline docs/lc_adapter.md:262]:
```suggestion
* `client_options.platform=carbon_black`: this indicates the data received will be Carbon Black events from their API.
```

File: docs/lc_adapter.md [hunk 8]
```diff
@@ -47,42 +77,231 @@ Other common configurations that are not required include:
...
+* `client_options.sensor_seed_key=....`: this is the value that identifies this instance of the Adapter. Record it to re-use the Sensor IDs generated for the CarbonBlack sensors from this Adapter later if you have to re-install the Adapter.
```
Comment (amrikr) [inline docs/lc_adapter.md:263]:
```suggestion
* `client_options.sensor_seed_key=....`: this is the value that identifies this instance of the Adapter. Record it to re-use the Sensor IDs generated for the Carbon Black sensors from this Adapter later if you have to re-install the Adapter.
```

File: docs/lc_adapter.md [hunk 9]
```diff
@@ -47,42 +77,231 @@ Other common configurations that are not required include:
...
+* `bucket_name:....`: the name of the S3 bucket holding the data.
+* `secret_key:....`: the API key for AWS that has access to this bucket.
+* `prefix=....`: the file/directory name prefix that holds the CarbonBlack data within the bucket.
```
Comment (amrikr) [inline docs/lc_adapter.md:266]:
```suggestion
* `prefix=....`: the file/directory name prefix that holds the Carbon Black data within the bucket.
```

**Review by amrikr — APPROVED** [review]
👍🏽

File: docs/lc_adapter.md (hunk 1 above)
Comment (maximelb) [reply docs/lc_adapter.md:14]: Thanks! Don't know what I was thinking. :)

**Pull request merged**